python/python/lancedb/query.py

```diff
@@ -267,6 +267,16 @@ def to_list(self) -> List[dict]:
         """
         return self.to_arrow().to_pylist()
 
+    def to_batches(self, max_chunksize) -> list:
+        """
+        Execute the query and return the results as a list of dictionaries.
+
+        Each list entry is a dictionary with the selected column names as keys,
+        or all table columns if `select` is not called. The vector and the "_distance"
+        fields are returned whether or not they're explicitly selected.
+        """
+        return self.to_arrow().to_batches(max_chunksize)
```

Comment:
in case you're looking for more help:

The `to_arrow` method will end up calling the `_execute_query` method here: https://github.com/lancedb/lancedb/blob/main/python/python/lancedb/table.py#L1517, which calls `LanceDataset.to_table`. This pulls the entire result set into memory.

Instead, i think you can do the following:

1. Change `_execute_query` to call `LanceDataset.to_batches()`
2. This new `to_batches` method will call `_execute_query` directly
3. Change `to_arrow()` to be `pa.Table.from_batches(self.to_batches())`


cc @wjones127 does that make sense?

Reply:
Thanks @changhiskhan I've tried `changing _execute_query` and `to_batches` to call `LanceDataset.to_batches() `and `_execute_query` respectively and rerunning the unit tests.

While executing `test_query_builder_batches` (with minor modifications) setting `batch_size = 1`, `number of rows = 2` and `limit = 2`, I expected two `RecordBatch` objects in the iterator, each containing 1 row but  I'm finding that the generator object of type `LanceScanner.to_batches` returned contains a single `RecordBatch` object with all (both) the rows.

This is the unit test passing currently:
```
def test_query_builder_batches(table):
    rs = (
        LanceVectorQueryBuilder(table, [0, 0], "vector")
        .limit(2)
        .select(["id", "vector"])
        .to_batches(1)
    )
    rs_list = []
    for item in rs:
        rs_list.append(item)
        assert isinstance(item, pa.RecordBatch)
    assert len(rs_list) == 1
    assert len(rs_list[0]["id"]) == 2
    assert all(rs_list[0].to_pandas()["vector"][0] == [1.0, 2.0])
    assert rs_list[0].to_pandas()["id"][0] == 1
    assert all(rs_list[0].to_pandas()["vector"][1] == [3.0, 4.0])
    assert rs_list[0].to_pandas()["id"][1] == 2
```
    
I was also able to reproduce this by calling the lance API directly on a table of 20 rows using
`res1 = ds.to_batches(columns=["vector", "id"], batch_size=1)`
here ds is a `LanceDataset`

Do we update the `LanceScanner` separately and make this change anyway?

Reply:
I don't think batch size is expected to work with vector search. There's no way to actually do the search without loading all results in memory. Batch size is more for scanning the whole table in chunks. So I think this is fine as is.